samples/wasi-threads/wasm-apps/CMakeLists.txt

```diff
@@ -1,6 +1,12 @@
 # Copyright (C) 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+if (APPLE)
+    set (HAVE_FLAG_SEARCH_PATHS_FIRST 0)
+    set (CMAKE_C_LINK_FLAGS "")
+    set (CMAKE_CXX_LINK_FLAGS "")
+endif ()
```

Comment:
why?

Reply:
it's the same as in the others `samples/<sample-name>/wasm-apps/CMakeLists.txt` files; in particular, without the `set (CMAKE_C_LINK_FLAGS "")` I get linking errors when compiling on a macbook:
```
wasm-ld: error: unknown argument: -search_paths_first
wasm-ld: error: unknown argument: -headerpad_max_install_names
```
I put the entire `if(APPLE)/endif()` block (and not only the `CMAKE_C_LINK_FLAGS` part) for consistency with the other samples

Reply:
ok. thank you for explanation.

i think it's cleaner to use ExternalProject as some of other samples do. but i guess it's ok for this PR.

Reply:
I'm not a big fan of external project because it doesn't automatically recompile sources from root on change. Maybe we can come up with a generic solution for all the samples, but that feels like outside of the scope for this pr

Reply:
i'm not a fan of ExternalProject either. but it's the cleanest way i'm aware of.

either ways, i agree it's nice to be consistent and it's out of the scope of this PR.